### ISSUE TITLE
Accessibility issue. Keep Inventory does not work in vaults.

### ISSUE THREAD

**Issue by PoyudiHytori** [Iskallia/Vault-public#821]
**Describe the bug**
Even with Keep Inventory set to on. Your inventory is still deleted when dying or failing a Vault run.

**To Reproduce**
1. Set Keep Inventory to true.
2. Die or run out of time in the vault.

**Expected behavior**
With Keep Inventory on you should not loose your items when dying for any reason.

**Info (please complete the following information):**
- World Type: Single player open to Lan.
- The_Vault Version [e.g. 1.0.1] 1.12.4

**Additional context**
I've been playing this mod-pack with friends. And we've been enjoying it. However due to this issue, one of my friends now refuses to enter the vault. And it's over-all had a greatly negative effect on our experience. The friends I play Minecraft with have Mental disorders. And in past Modpacks the KeepInventory gamerule has been a life-line that has allowed them to enjoy playing minecraft. To take risks, adventure, explore. Etc. With this gamerule not working for the vaults, it has quickly become too triggering for them to function inside of the vaults. I'd like to enjoy this mod-pack with my friends, and I'd like them to have an enjoyable time playing it. So I'm hopeing this is indeed a bug. And not an intended punishment built into the Vault Mod.

**Comment by bmemike** [comment]
keepInventory "not working" is functioning exactly as designed. Vault deaths are meant to result in you losing your items.

It's a high risk, high reward pack.

There are two ways to modify this. 

If you apply a Soul Flame to a crystal it'll grant the Afterlife effect to it. With Afterlife if you die in the vault you will be able to retain everything that you entered the vault with (but nothing you gathered in the vault). 

The second is the Phoenix modifer. It mostly functions the same except you can apply this to vault crystals with catalysts. 

Sorry to hear about the experience your friend had, but this is entirely intended behavior and vault deaths (and recovering from them) is expected.

**Comment by PoyudiHytori** [comment]
So your saying. Instead of using KeepInventory. A gamerule In the game. I should go into Creative Mode. Spawn in a bunch of items. Cheat like crazy. All to get the same effect? And that's "Intended Behavior"?

**Comment by Steakchest** [comment]
No, it isn't what he's saying. He's saying you losing your stuff in the vaults is how the game is intended to be played. KeepInventory is not. Either way, you're changing the gameplay mechanics for an advantage, or cheating. This isn't a bug, it's how the game is supposed to work.

**Comment by bmemike** [comment]
That's not entirely what I'm saying either.

What I'm saying is:
* keepInventory not working in the vault is intended behavior
* the vault provides mechanisms to reduce the impact of doing that (Afterlife and Phoenix, as mentioned)
* those items can be crafted and you have access to them without "cheating like crazy"

**Comment by PoyudiHytori** [comment]
As far as I know. What is keeping KeepInventory from working was **not** added to the mod to keep players from using keep inventory. But rather it was put in because players were able to teleport back into the vault to recover their items. And so they made the game wipe your inventory clean to prevent that exploit, probably as a way to keep people from exploiting this in servers. Making it so KeepInventory didn't work in the vaults was an unintended side-effect of that. Looking through the bug reports, the website, curse forge, and reddit. I was unable to find a place where there was any official word about this bug from the Devs. So I have no idea where this idea of it being "Intended" to not work like this is coming from. If you have some kind of official source for this. Please do share.

Me and my friends are playing this pack in our own world. Playing it our way. And making it fun for us. Us being able to use KeepInventory would not effect players who don't want to use it. I don't see why it matters to so many people how we choose to play this modpack, or minecraft in general.

As for those "Mechanisms" Using Afterlife would mean we could never get to the final vault, or actually finish the modpack. Which I'm sure isn't intended. And to get Phoenix on every vault, to avoid not being able to complete the modpack, would entail spawning in a ton of items, to keep rerolling, until we finally make a crystal with that modifier. Since it is highly RNG dependent. Those Mechanisms, to me at least, do not seem like they are meant to replace KeepInventory. Rather they are for players who play without KeepInventory, or for servers. Afterlife being meant for the beginning of the game to help you learn the vaults and get a footing with less risk involved for your character. And Phoenix being a rare modifier that let's a player run through a vault in a more risky fashion than normal. Not to be used on every vault there is. They are not replacements for KeepInventory. They are not "Just as good". They don't even appear to be meant to be used that way.

So yes. In order to complete the mod-pack using those Mechanisms instead of KeepInventory. Would mean that me and my friends would have to cheat, a considerable amount of time. By either spawning in the artifacts when we feel like we want to take on the final vault. Or by spawning in a lot of crystals and Catalysts To re-roll for the Phoenix Modifier every time we want to take on a vault.
